src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs

```diff
@@ -84,6 +101,17 @@ public int ExecuteAsync()
             }
         }
 
+        internal void OnTestSessionTimeout(object obj)
+        {
+            string message = String.Format(ClientResources.TestSessionTimeoutMessage, this.testSessionTimeout);
```

Comment:
Knowledge of json serializer and that we need to log to two instances (regular and rawmessage) has now percolated to TestRunRequest. Is there a better way to hide this complexity?